src/lib/ops.py

```diff
@@ -195,12 +201,12 @@ def Rotation(v: np.ndarray, theta: float) -> np.ndarray:
   """Produce the single-qubit rotation operator."""
 
   v = np.asarray(v)
-  if (v.shape != (3,) or not np.isclose(v @ v, 1) or
+  if (v.shape != (3,) or not math.isclose(v @ v, 1) or
```

Comment:
Lesson learned: `math.isclose()` is 300 times faster than `np.isclose()` for scalars:
```
$ python3 -m timeit -s "import numpy as np; import math" "np.isclose(1.0001, 1)"                                                                                                                                                                       
20000 loops, best of 5: 15.4 usec per loop
$ python3 -m timeit -s "import numpy as np; import math" "math.isclose(1.0001, 1)"                                                                                                                                                                     
5000000 loops, best of 5: 48.4 nsec per loop
```

See: https://github.com/numpy/numpy/issues/16160 .